index.js

```diff
@@ -121,22 +121,30 @@ app.listen(3001, () => {
   console.log("Server running on port 3001");
  });
 
+ var receiptCalls = [];
 
-const processTxData = (rawData) => {
+const processTxData = async (rawData) => {
   console.log('>> Attempting to process data..');
   processedData = [...rawData.transactions];
   console.log('>> Processed data [0] = ',processedData[0]);
 
-  processedData.map((tx) => {
-    tx.receipt = await pullTxReceipt(tx.ethHash)
+
+
+  processedData = processedData.map((tx) => {
+    receiptCalls.push(tx.receipt = pullTxReceipt(tx.ethHash))
   });
```

Comment:
avoid mutating references or reassigning global references and instead create a new one, memory should not be a concern and if it becomes one; you can always start by removing processed data and spreading directly to the .map call
```suggestion
  const result = await Promise.all(processedData.map(async (tx) => {
    tx.receipt = await pullTxReceipt(tx.ethHash);
    return tx;
  }));
  return result;
```